src/sqlfluff/core/linter/linter.py

```diff
@@ -126,7 +126,7 @@ def load_raw_file_and_config(
         config_encoding: str = file_config.get("encoding", default="autodetect")
         encoding = get_encoding(fname=fname, config_encoding=config_encoding)
         # Check file size before loading.
-        limit = file_config.get("large_file_skip_byte_limit")
+        limit = int(file_config.get("large_file_skip_byte_limit"))
```

Comment:
I like the approach, but at the moment this will probably just introduce another hard to ready error if someone puts in the wrong value for the config.

In particular, if the value isn't set then we'll get an error because `int(None)` is a `ValueError` think.